### PR TITLE
Loads plugins and notiifies them on status for each service.

### PR DESCRIPTION
**Pull request by sreeix**
Here is a small patch that loads and sends notifications to the plugins. 
Plugins are loaded from api.js. and are stored in the plugins folder.
the notifications currently support up/down/unknown/critical (status update every refresh)

We could also just hook to refresh to get one big status update.
This would pave the way for interesting usages. twitter notification is one i am thinking about now.

Thoughts?

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -103,10 +119,12 @@ var commands = {
         service.status = "up";
         service.statusCode = 0;
         service.message = "";
+        controller.emit(up, service);
```
Comment (obazoud) [inline api.js]: Shoud be "up" instead of up ?
Why don't you use service.status ?

File: api.js [hunk 2]
```diff
@@ -157,12 +177,14 @@ var commands = {
...
       service.statusCode = status;
       service.message = message;
+      controller.emit('down', service);
     };
 
     stream.addListener('error', function (e) {
       service.status = "down";
       service.statusCode = 0;
       service.message = e.message;
+      controller.emit('error', service);
```
Comment (obazoud) [inline api.js]: Shoud be "down" instead of 'error' ?
Why don't you use service.status ?

**Comment by obazoud** [comment]
It's is ok for me.
Could you fix the pull request code, see my review, and maybe send me a unique pull request ?

**Comment by sreeix** [comment]
Fixed the errors that you mention in the review. I also use service.status for all the status updates.

**Pull request merged**